paasta_tools/cli/cmds/rollback.py

```diff
@@ -19,8 +22,40 @@
 from paasta_tools.cli.utils import lazy_choices_completer
 from paasta_tools.cli.utils import list_services
 from paasta_tools.generate_deployments_for_service import get_instance_config_for_service
+from paasta_tools.remote_git import list_remote_refs
+from paasta_tools.utils import datetime_from_utc_to_local
+from paasta_tools.utils import format_table
 from paasta_tools.utils import get_git_url
 from paasta_tools.utils import PaastaColors
+from paasta_tools.utils import parse_timestamp
+
+
+def list_previously_deployed_shas(parsed_args, **kwargs):
+    service = parsed_args.service
+    deploy_groups = set([deploy_group for deploy_group in parsed_args.deploy_groups.split(',') if deploy_group])
+    return (sha for sha, _ in get_git_shas_for_service(service, deploy_groups))
+
+
+def get_git_shas_for_service(service, deploy_groups):
+    """Returns a list of 2-tuples of the form (sha, timestamp) for each deploy tag in a service's git
+    repository"""
+    if service is None:
+        return []
+    git_url = get_git_url(service=service)
+    all_deploy_groups = set(config.get_deploy_group() for config in get_instance_config_for_service(
+        soa_dir=DEFAULT_SOA_DIR,
+        service=service,
+    ))
+    deploy_groups, _ = validate_given_deploy_groups(all_deploy_groups, deploy_groups)
+    regex = r'^refs/tags/(?:paasta-){1,2}(?:%s)-(?P<tstamp>\d{8}T\d{6})-deploy$' % '|'.join(deploy_groups)
```

Comment:
This regex man, is totally going to bite us.

How hard would it be to keep this regex in one place? A function like "extract_tags", and then in this function just grab the tags you need, etc?